merlin-sdk/src/main/java/gov/nasa/jpl/aerie/merlin/protocol/model/ConfigurationType.java

```diff
@@ -17,5 +17,21 @@ Config instantiate(Map<String, SerializedValue> arguments)
   Map<String, SerializedValue> getArguments(Config configuration);
   List<String> getValidationFailures(Config configuration);
 
-  final class UnconstructableConfigurationException extends Exception {}
+  final class UnconstructableConfigurationException extends Exception {
+    public UnconstructableConfigurationException() {
+      super();
+    }
+
+    public UnconstructableConfigurationException(final String message) {
+      super(message);
+    }
+
+    public static UnconstructableConfigurationException unconstructableArgument(final String argumentName, final String failure) {
+      return new UnconstructableConfigurationException("Unconstructable argument \"%s\": %s".formatted(argumentName, failure));
+    }
+
+    public static UnconstructableConfigurationException nonexistentArgument(final String argumentName) {
+      return new UnconstructableConfigurationException("Nonexistent argument \"%s\"".formatted(argumentName));
```

Comment:
As noted in the future work section it would probably be best to split this exception up. This will further impact the mission modeler test framework (since they must handle these exceptions) – so this may require a bit of design work to abstract those exceptions away for the test framework.